source/containers.tex

```diff
@@ -214,11 +214,9 @@
  constant                   \\ \rowsep
 
 \tcode{i <=> j}             &
-  \tcode{strong_ordering}
-  if \tcode{X::iterator} meets the random access iterator requirements,
-  otherwise \tcode{strong_equality} &
-                             &
-                             &
+ \tcode{strong_ordering}    &
+                            &
+ \expects \tcode{X::iterator} meets the random access iterator requirements &
```

Comment:
`\expects` seems wrong to me here: this requires all container iterators to implement `<=>` despite that it would be UB to evaluate `i <=> j` when `i` and `j` are iterators for non-random-access containers. Given that we have concepts that detect comparability based on the validity of expressions, it's not good design to have unusable expressions be well-formed.

I think we should handle this bit of cleanup via LWG issue rather than editorially.

Reply:
I think you are right. And there might 2 kinds of solution IMO, which is preferred?

1. use `\mandates` (or `\constraints`?) instead of `\expects`;
2. make a new requirement called "random access container" and move `i <=> j` to its own table.

Reply:
Please be so kind and leave the LWG issue number in a comment here.

Reply:
I've submitted this issue to LWGChair, I'll reply back with an issue number once it's assigned.

Reply:
This is now [LWG 3352](https://cplusplus.github.io/LWG/issue3352).